### PR TITLE
replace default images

### PR DESCRIPTION
**Pull request by sharbuz**
## What is the problem this PR solves?

Jenkins->Buildkite pipelines migration 

Replace default images by our own images

Example of the pipelines:

Examples of the affected pipelines:

## Related issues
https://github.com/elastic/ingest-dev/issues/1693

### CODE REVIEW

File: .buildkite/hooks/pre-command
Comment (alexsapran) [inline .buildkite/hooks/pre-command:95]: @dliappis I wonder if we could set this in our Windows images, so we don't need to do anything in the repo scripts.

Happy to see if I can make it work for windows.

Comment (dliappis) [reply .buildkite/hooks/pre-command:95]: @alexsapran yup this would be an easy one to fix. I suppose we should add it at the global level?

Comment (sharbuz) [reply .buildkite/hooks/pre-command:95]: > @alexsapran yup this would be an easy one to fix. I suppose we should add it at the global level?

In my opinion - yes

Comment (sharbuz) [reply .buildkite/hooks/pre-command:95]: I'll remove that when it's migrated into our images

Comment (alexsapran) [reply .buildkite/hooks/pre-command:95]: I have raised a PR https://github.com/elastic/ci-agent-images/pull/589
Hopefully, soon it will be merged.

Comment (alexsapran) [reply .buildkite/hooks/pre-command:95]: I have rebuild and pushed a new image, using the latest image should be enough and this is not required anymore

Comment (sharbuz) [reply .buildkite/hooks/pre-command:95]: > I have rebuild and pushed a new image, using the latest image should be enough and this is not required anymore

@alexsapran it doesn't work as expected.
There are examples issues:
https://buildkite.com/elastic/beats-metricbeat/builds/4088
https://buildkite.com/elastic/beats-xpack-filebeat/builds/709
https://buildkite.com/elastic/beats-xpack-metricbeat/builds/1607
There are result of using commands in the pre-command:
https://buildkite.com/elastic/beats-metricbeat/builds/4078
https://buildkite.com/elastic/beats-xpack-filebeat/builds/699
https://buildkite.com/elastic/beats-xpack-metricbeat/builds/1597

Comment (alexsapran) [reply .buildkite/hooks/pre-command:95]: Thank @sharbuz I found the problem, which was the fact the BK was running with a specific user. It should be ok now.

Comment (alexsapran) [reply .buildkite/hooks/pre-command:95]: retrying to a previous failing build now works 
https://buildkite.com/elastic/beats-metricbeat/builds/4088#_

File: .buildkite/scripts/setenv.sh
Comment (alexsapran) [inline .buildkite/scripts/setenv.sh:5]: Maybe we should remove this, leaving a comment so I don't forget about it.

File: .buildkite/hooks/pre-command
Comment (alexsapran) [inline .buildkite/hooks/pre-command:16]: Why do we need to do this now? This is handled inside the VM image.

File: .buildkite/scripts/generate_auditbeat_pipeline.sh
Comment (alexsapran) [inline .buildkite/scripts/generate_auditbeat_pipeline.sh:28]: I see some places where we do a single `*.*` and others with explicit paths in the pipeline files.

File: .buildkite/scripts/generate_filebeat_pipeline.sh
Comment (alexsapran) [inline .buildkite/scripts/generate_filebeat_pipeline.sh:1]: How are we adding this file? Didn't Filebeat already exist?

File: .buildkite/scripts/generate_filebeat_pipeline.sh
Comment (alexsapran) [inline .buildkite/scripts/generate_filebeat_pipeline.sh:174]:
```suggestion
      - label: ":windows: Windows 11 Unit Tests"
```

The label in the UI feels strange, so adding the prefix IMHO makes it more precise. The same applies to the other Windows steps.

File: .buildkite/scripts/generate_filebeat_pipeline.sh
Comment (alexsapran) [inline .buildkite/scripts/generate_filebeat_pipeline.sh:219]: This script internally creates two steps. That internally runs the `mage package,` but it also publishes this to our Docker registry, right?

I don't know if this is the right PR to solve this, but I would like to review it now if we are doing something wrong.

**Comment by pazone** [comment]
LGTM. 
This PR contains a lot of work and makes big progress. 
To reviewers: Please make sure that you leave comments about only critical findings. Let's move it forward

File: .buildkite/scripts/generate_auditbeat_pipeline.sh
Comment (sharbuz) [reply .buildkite/scripts/generate_auditbeat_pipeline.sh:28]: fixed

File: .buildkite/scripts/generate_xpack_packetbeat_pipeline.sh
Comment (alexsapran) [inline .buildkite/scripts/generate_xpack_packetbeat_pipeline.sh:55]: we might have missed something, this should be similar to https://github.com/elastic/beats/blob/main/x-pack/packetbeat/Jenkinsfile.yml#L91

```suggestion
		command: |
          Set-Location -Path $BEATS_PROJECT_NAME
          mage build unitTest
```

**Review by alexsapran — COMMENTED** [review]
Overall, it looks good to me +1. I only noticed one Windows step that had the wrong command. 
I synced with @pazone to investigate the package step. If there is an issue with that step, I would like to address it before launching this massive PR. 
@pazone, please approve it once you know so we can merge it.

File: .buildkite/scripts/generate_xpack_packetbeat_pipeline.sh
Comment (alexsapran) [reply .buildkite/scripts/generate_xpack_packetbeat_pipeline.sh:55]: While we should be doing build and unitTests, the build was run successfully, so I guess its ok?
https://buildkite.com/elastic/beats-xpack-packetbeat/builds/1841#018ea97a-af2f-44af-8107-cda3dc56de05

**Comment by pazone** [comment]
Looks like the package steps can be removed for PRs. 
I propose to merge this PR and I can disable those steps later.

**Review by pazone — APPROVED** [review]

File: .buildkite/scripts/generate_xpack_packetbeat_pipeline.sh
Comment (sharbuz) [reply .buildkite/scripts/generate_xpack_packetbeat_pipeline.sh:64]: I think so

File: .buildkite/scripts/generate_filebeat_pipeline.sh
Comment (sharbuz) [reply .buildkite/scripts/generate_filebeat_pipeline.sh]: fiexed

**Comment by pazone** [comment]
@sharbuz has changed packaging to `mage package` and now we don't publish artifacts on PRs

File: .buildkite/scripts/generate_filebeat_pipeline.sh
Comment (sharbuz) [reply .buildkite/scripts/generate_filebeat_pipeline.sh]: fixed

**Pull request merged**
